MAVProxy/modules/mavproxy_paramedit/param_editor.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+'''
+param editor module
+Akshath Singhal
+June 2019
+'''
+
+from MAVProxy.modules.lib import mp_util
+from MAVProxy.modules.lib import multiproc
+from MAVProxy.modules.mavproxy_paramedit import ph_event
+import time
+import threading
+from MAVProxy.modules.lib import wx_processguard
+from ..lib.wx_loader import wx
+from MAVProxy.modules.mavproxy_paramedit import param_editor_frame
+from pymavlink import mavutil
+
+ParamEditorEvent = ph_event.ParamEditorEvent
+
+
+class ParamEditorEventThread(threading.Thread):
+    def __init__(self, mp_paramedit, queue, lock):
+        threading.Thread.__init__(self)
+        self.mp_paramedit = mp_paramedit
+        self.event_queue = queue
+        self.event_queue_lock = lock
+        self.time_to_quit = False
+        self.queue_access_timeout = 0.5
+
+    def module(self, name):
+        '''access another module'''
+        return self.mp_paramedit.mpstate.module(name)
+
+    def master(self):
+        '''access master mavlink connection'''
+        return self.mp_paramedit.mpstate.master()
+
+    def run(self):
+        while not self.time_to_quit:
+            queue_access_start_time = time.time()
+
+            while (self.event_queue.qsize() > 0):
```

Comment:
I believe `qsize()` only gives you an estimate of the queue size.  You should check you actually did pop an item off in the get call below.

Reply:
This is used as a secondary check. The events are extracted using non-blocking get method and try-exception block handles well if there was no actual content to be popped.